helix-view/src/graphics.rs

```diff
@@ -92,6 +92,55 @@ impl Rect {
         self.y.saturating_add(self.height)
     }
 
+    // Returns a new Rect with width reduced from the left side.
+    // This changes the `x` coordinate.
+    pub fn chop_from_left(self, width: u16) -> Rect {

```

Comment:
Let's call these `shrink_`?

Reply:
I initially called it `shrink_`, but it seemed to make it sound like the area _itself_ shrinks to the specified width instead of `width` amount of area being cut off from it. `chop_` sounds a bit... childish(?) maybe, but I think it gets the intent across pretty well.

Reply:
\cc @cessen for naming ideas

Reply:
Not sure I have any great naming ideas here.

We could try naming it based on use-case.  In particular, I suspect this would be used when you want a sub-rectangle of an existing `Rect`, behaving kind of like a child.  So it could be something like `sub_rect_left()`, where `left` indicates the side that's being kept rather than removed.  (Although that might get confusing with "sub" also being a common abbreviation for "subtract".)

Other possibilities more along the lines of "chop":

* `reduce_from_left()`
* `clip_left()` ("clip" is pretty common graphics terminology, but maybe people working on a text editor won't be familiar)

I'm honestly not sure what would be best.  This might be a case where we just need to wait until we forget about it and encounter it at a use-site, and see if it ends up being confusing or not.

Reply:
What about `shrink_from_bottom` and `shrink_bottom_to`?

Reply:
I kind of like `clip` or `shrink_from_`

Reply:
“crop” is a popular image editing term that might or might not fit here, just mentioning it since nobody else did yet (although “clip” was)

Reply:
I think `clip` fits quite nicely too.